index.js

```diff
@@ -1,95 +1,125 @@
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
+function drinkWater(currentThirstLevel) {
+    console.log('Man I sure am thirsty')
+    currentThirstLevel -= 1
+    console.log('Ahh that hits the spot')
+    return currentThirstLevel
+}
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum) {
+  if (num === maybeNum) {
+    return 'same same';
+  }
+  else if (num == maybeNum) {
+    return 'same same (but different)'
+  }
+  else {
+    return 'different'
+  }
+}
+
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
+
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
-//
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
-//
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
-//
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
+
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
+console.log("%cQuestion 5", "color: red")
+
+function updateGrade(student, grade) {
+  student.grade = grade
+}
 
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)  
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red") 
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)  
+// => { name: "Duane", grade: 92 }
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 6 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
-// // => "Duane"
-// // => "Cell: 555-123-4567"
-// // => "Office: 555-456-7890"
-// // => "Liza"
-// // => "Cell: 555-234-5678"
-// // => "Office: 555-567-1234"
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 6", "color: red")
+
+function printNameAndPhones(users) {
+  for (user in users) {
+    console.log (users[user]['name']);
+    console.log (`Cell: ${users[user]['phones']['cell']}`);
+    console.log (`Office: ${users[user]['phones']['office']}`);
+  }
```

Comment:
`for..in` works, but it's better to use `for..of` or `forEach` when you're working with arrays. Also, try using the dot notation instead of bracket notation - it's more convention in Javascript to use dot notation on objects. 

Here's how it would look with `for..of`: 

```js
  for (let user of users) {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  }
```

Here's how it would look with `forEach`: 

```js
  users.forEach(function(user) {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  })
```